spec/requests/muted_users_n_plus_one_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "n+1 queries in the muted users controller" do
+  include LoginMacros
+
+  describe "#index" do
+    context "with a logged in user who has muted someone", n_plus_one: true do
+      let!(:muter) { create(:user) }
+
+      populate do |n|
+        muted_users = create_list(:user, n)
+        muted_users.each do |muted|
+          Mute.create(muter: muter, muted: muted)
+        end
+      end
+
+      before do
+        fake_login_known_user(muter)
+      end
+
+      subject do
+        proc do
+          get user_muted_users_path(user_id: muter)
+        end
+      end
+
+      warmup { subject.call }
+
+      it "produces a constant number of queries" do
+        expect { subject.call }
+          .to perform_constant_number_of_queries
```

Comment:
This needs the `with_attached_*` here: https://github.com/otwcode/otwarchive/blob/1588f6c552b1c85adeb1b30e31411cb2cb6f37d7/app/controllers/muted/users_controller.rb#L14